hllplus/sparse.go

```diff
@@ -289,6 +289,11 @@ func (s *deltaSlice) Clone() *deltaSlice {
 }
 
 func (s *deltaSlice) Append(x uint32) {
+	// duplicate element:
```

Comment:
@mxmCherry I understand the problem, but it's the wrong place to fix it

Reply:
the right place to fix it is in `func (s *sparseState) Flush()`, it should be this:

```go
// merge existing data and buffered
s.data.Iterate(func(x uint32) {
  if len(buffered) == 0 {
    result.Append(x)
    return
  }

  for {
    if u := buffered[0]; u < x {
      result.Append(u)
      buffered = buffered[1:]
    } else {
      result.Append(x)
      break
    }
  }
})
```

Reply:
That doesn't solve it.

If you add the same element, like `.Add(1); .Add(1)` it will be duplicated.

I'll try to squeeze the fix here, but it'll be not as accurate as doing it in deltaSlice (but I agree, deltaSlice is not the most expected place to "deduplicate").

Reply:
Roughly:

```go
		subject.Add(1)
		subject.Flush()
		subject.Add(1)
		subject.Flush()
		subject.Add(1)
		subject.Flush()
		subject.Add(1)
		subject.Flush()
		subject.Add(1)
		subject.Flush()
		Expect(subject.data.Count()).To(BeNumerically("==", 1)) // fails with actual=5
```